llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2365,9 +2365,69 @@ static bool isNonZeroAdd(const APInt &DemandedElts, unsigned Depth,
       .isNonZero();
 }
 
+// Check to see if A is both a GEP and is the incoming value for a PHI in the
```

Comment:
Can you please pre-commit the move of this function?

Reply:
Actually, I don't think we need this. Can you please rebase this on top of https://github.com/llvm/llvm-project/commit/337504683efa0b898bcf69b4e5be67d73dbaf180 and then add the ptrtoint handling inside isKnownNonEqual()? I think that is the cleanest way to handle this.

isKnownNonZero for sub will call isKnownNonEqual, that will look through ptrtoint and call itself recursively, and at that point we'll handle the GEP case. This will make sure that if we add any extra pointer cases, we'll automatically make use of them as well.

Reply:
Agreed. This is a good improvement. Thanks for the suggestion.